db/migrate/20171227194024_create_moderator_actions.rb

```diff
@@ -0,0 +1,12 @@
+class CreateModeratorActions < ActiveRecord::Migration[5.1]
+  def change
+    create_table :moderator_actions, id: :uuid do |t|
+      t.string :action
+      t.string :reason
+      t.references :moderator, type: :uuid, index: true
+      t.references :moderatable, type: :uuid, index: true
+      t.string :moderatable_type
```

Comment:
I'm not super happy with this, since polymorphic can go overboard; however having a single cohesive moderator activity log across all models seems reasonable to me; so I'm OK with it for now. Thoughts?